src/tests/Microsoft.PowerFx.Interpreter.Tests/LanguageServiceProtocol/LanguageServerTests.cs

```diff
@@ -446,12 +446,15 @@ private void CheckBehaviorError(string sentToClientData, bool expectBehaviorErro
         }
 
         [Theory]
-        [InlineData("Color.AliceBl", 13, false)]
-        [InlineData("Behavior(); Color.AliceBl", 25, true)]
+        [InlineData("Color.AliceBl", 13, false, true)]
+        [InlineData("Behavior(); Color.AliceBl", 25, true, true)]
 
         // $$$ This test generates an internal error as we use an behavior function but we have no way to check its presence
-        [InlineData("Behavior(); Color.AliceBl", 25, false)]
-        public void TestCompletion(string text, int offset, bool withAllowSideEffects)
+        [InlineData("Behavior(); Color.AliceBl", 25, false, true)]
+        [InlineData("Color.AliceBl", 13, false, false)]
+        [InlineData("Behavior(); Color.AliceBl", 25, true, false)]
+        [InlineData("Behavior(); Color.AliceBl", 25, false, false)]
```

Comment:
Looks like you're duplicating all tests
What about: 
foreach (bool addExprToUri in new[] { true, false })
{ 
...
}

Reply:
Sure the code is duplicate but they test different scenarios. Duplicated test checks the expression being different in uri and in the new "Text" property and "Text" property is preferred. So if expression are present both in uri and Text property but they are different and a completion operation is performed for example, then results for the expressio in "Text" property is returned instead of the one in uri

Reply:
Refactored to Avoid duplication